lib/flux-core/src/iter/adapters/skip.rs

```diff
@@ -0,0 +1,10 @@
+use flux_attrs::*;
+
+#[extern_spec(core::iter)]
+#[refined_by(size: int)]
+struct Skip<I>;
+
+#[extern_spec(core::iter)]
+#[assoc(fn size(r: Skip) -> int { r.size } )]
+#[assoc(fn step(self: Skip, other: Skip) -> bool { other.size == self.size - 1 } )]
+impl<I: Iterator> Iterator for Skip<I> {}
```

Comment:
Same as `Map`, this shouldn't work without annotating `next`

Reply:
done